### DISCUSSION TITLE
The abnormal status of the child process after executing

### DISCUSSION THREAD

**Discussion by Zhang-pchao** [deepmodeling/dpgen#682] [Q&A]
The job management system I use is a modified version of PBS. I modified some of the code in PBS.py, but I encountered the following error, is it the reason for the abnormal status of the child process after executing it, and how can I solve it? thanks.

  File "/my_path/dpgen/dispatcher/Batch.py", line 123, in submit
    self.do_submit(job_dirs, cmd, args, res, outlog=outlog, errlog=errlog)
  File "/my_path/dpgen/dispatcher/PBS.py", line 56, in do_submit
    stdin, stdout, stderr = self.context.block_checkcall('cd %s && %s %s' % (self.context.remote_root, 'qsub', self.sub_script_name))
  File "/my_path/dpgen/dispatcher/LocalContext.py", line 153, in block_checkcall
    raise RuntimeError("Get error code %d in locally calling %s with job: %s ", (code, cmd, self.job_uuid))
RuntimeError: ('Get error code %d in locally calling %s with job: %s ', (1, 'cd /work_path/tmp/train/job_id && qsub job_id.sub', 'job_id'))

dpgen/dispatcher/LocalContext.py", line 153:
    def block_call(self, cmd) :
        cwd = os.getcwd()
        os.chdir(self.remote_root)
        proc = sp.Popen(cmd, shell=True, stdout = sp.PIPE, stderr = sp.PIPE)
        o, e = proc.communicate()       
        stdout = SPRetObj(o)
        stderr = SPRetObj(e)
        code = proc.returncode
        os.chdir(cwd)        
        return code, None, stdout, stderr

**Comment by wanghan-iapcm** [comment]
Please switch to the remote fold to check if the submit script generated by dpgen is correct. 

It is recommended to use the [dpdispatcher](https://github.com/deepmodeling/dpdispatcher) over the dispatcher module of dpgen.
